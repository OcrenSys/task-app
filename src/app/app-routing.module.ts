import { NgModule } from '@angular/core';
import { PreloadAllModules, RouterModule, Routes } from '@angular/router';
const routes: Routes = [
  {
    path: '',
    loadComponent: () =>
      import('./pages/list/list.component').then((c) => c.ListComponent),
  },

  {
    path: 'details',
    loadComponent: () =>
      import('./pages/details/details.component').then(
        (c) => c.DetailsComponent
      ),
  },
  {
    path: '',
    redirectTo: '',
    pathMatch: 'full',
  },
];

@NgModule({
  imports: [
    RouterModule.forRoot(routes, { preloadingStrategy: PreloadAllModules }),
  ],
  exports: [RouterModule],
})
export class AppRoutingModule {}
