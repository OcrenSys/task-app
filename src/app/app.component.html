<app-navbar></app-navbar>

<div class="content" role="main">
  <div class="head">
    <button mat-raised-button color="primary" (click)="openDialog()">
      <mat-icon
        aria-hidden="false"
        aria-label="Example home icon"
        fontIcon="add"
      ></mat-icon>
      Nueva Tarea
    </button>
  </div>

  <app-task-list></app-task-list>
</div>
