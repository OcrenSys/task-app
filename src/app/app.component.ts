import { Component } from '@angular/core';
import { MatDialog } from '@angular/material/dialog';
import { TaskFormComponent } from './core/components/task-form/task-form.component';
@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
})
export class AppComponent {}
