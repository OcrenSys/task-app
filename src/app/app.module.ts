import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { ListComponent } from './pages/list/list.component';
import { DetailsComponent } from './pages/details/details.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { NavbarComponent } from './core/components/navbar/navbar.component';
import { TaskListComponent } from './core/components/task-list/task-list.component';
import { TaskFormComponent } from './core/components/task-form/task-form.component';
import { MatDialogModule } from '@angular/material/dialog';
import { MatCardModule } from '@angular/material/card';

@NgModule({
  declarations: [AppComponent, ListComponent, DetailsComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    MatButtonModule,
    MatIconModule,
    NavbarComponent,
    TaskListComponent,
    TaskFormComponent,
    MatDialogModule,
    MatCardModule,
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
