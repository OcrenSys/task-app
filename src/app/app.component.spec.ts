import { TestBed, ComponentFixture } from '@angular/core/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(() =>
    TestBed.configureTestingModule({
      declarations: [AppComponent],
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  test('should create the app', () => {
    expect(component).toBeTruthy();
  });

  test('should make match with the snapshot', () => {
    const compile = fixture?.nativeElement as HTMLElement;
    expect(compile).toMatchSnapshot();
  });
});
