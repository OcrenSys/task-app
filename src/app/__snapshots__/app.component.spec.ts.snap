// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should make match with the snapshot 1`] = `
<div
  id="root1"
>
  <app-navbar />
  <div
    class="content"
    role="main"
  >
    <router-outlet />
  </div>
</div>
`;
